spec/haproxy/templates/haproxy_config/frontend_https_spec.rb

```diff
@@ -88,23 +98,43 @@
         default_properties.merge({ 'client_cert' => 'client_cert contents', 'client_cert_ignore_err' => true })
       end
 
-      # FIXME: if client_cert_ignore_err is true but client_cert is not provided, then it should error
-
       it 'adds the crt-ignore-err flag' do
         expect(frontend_https).to include('bind :443  ssl crt /var/vcap/jobs/haproxy/config/ssl  ca-file /etc/ssl/certs/ca-certificates.crt verify optional crt-ignore-err true')
       end
+
+      context 'when client_cert is not provided' do
+        let(:properties) do
+          default_properties.merge({ 'client_cert_ignore_err' => true })
+        end
+
+        it 'aborts with a meaningful error message' do
+          expect do
+            frontend_https
+          end.to raise_error /Conflicting configuration: must provide client_cert to use client_cert_ignore_err/
+        end
+      end
     end
 
     context 'when ha_proxy.client_revocation_list is provided' do
       let(:properties) do
         default_properties.merge({ 'client_cert' => 'client_cert contents', 'client_revocation_list' => 'client_revocation_list contents' })
```

Comment:
`client_cert` is actually a boolean flag. so the value should be `true` or `false` rather than a string.

See https://github.com/cloudfoundry-incubator/haproxy-boshrelease/blob/master/jobs/haproxy/spec#L317

Reply:
Fixed